api/applications_api.py

```diff
@@ -57,39 +43,97 @@ def perform_create(self, serializer):
 
 
 class OrganisationSerializer(serializers.ModelSerializer):
+
+    id = serializers.IntegerField(allow_null=True, required=False)
+
+    def to_internal_value(self, data):
```

Comment:
Here is some kind of proof of concept code, which may or may not be useful. When using writing nested serializers uniqueness can't be determined by rest framework automatically it seems, so we skip uniqueness check when id matches the one in database.